service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/CreateAiNotebookInstanceStepTest.java

```diff
@@ -31,6 +54,79 @@ public class CreateAiNotebookInstanceStepTest extends BaseUnitTest {
+  @Mock private OperationCow.OperationAdapter.StatusAdapter mockOperationStatusAdapter;
+  @Mock private Operation mockOperation;
+
+  private static final String FAKE_PROJECT_ID = "fakeprojectid";
+
+  // The permission denied code includes cases of the "quota exceeded limit" error.
+  @Test
+  public void create_throwsDuringOperationPolling_Permission_Denied() throws IOException {
+    // Create the mock AIPlatform Notebook cow,
+    Mockito.doReturn(mockAIPlatformNotebookCow).when(mockCrlService).getAIPlatformNotebooksCow();
+    when(mockAIPlatformNotebookCow.instances()).thenReturn(mockInstances);
+    when(mockInstances.create(any(InstanceName.class), any(Instance.class)))
+        .thenReturn(mockInstancesCreate);
+    when(mockInstancesCreate.execute()).thenReturn(mockOperation);
+
+    when(mockAIPlatformNotebookCow.operations()).thenReturn(mockOperations);
+
+    // And mock the OperationCow.
+    when(mockOperations.operationCow(any(Operation.class))).thenReturn(mockOperationCow);
+    when(mockOperationCow.getOperationAdapter()).thenReturn(mockOperationAdapter);
+    when(mockOperationAdapter.getDone()).thenReturn(true);
+    when(mockGcpCloudContextService.getRequiredGcpProject(any())).thenReturn(FAKE_PROJECT_ID);
+
+    // Create fake configs for the create notebook step.
+    CliConfiguration fakeCliConfiguration = new CliConfiguration();
+    fakeCliConfiguration.setServerName("fake-server-name");
+
+    VersionConfiguration fakeVersionConfig = new VersionConfiguration(new MockEnvironment());
+    fakeVersionConfig.setGitHash("fake-git-hash");
+
+    // Mock the create step.
+    final CreateAiNotebookInstanceStep createNotebookStep =
+        new CreateAiNotebookInstanceStep(
+            ControlledResourceFixtures.makeDefaultAiNotebookInstance().build(),
+            "fake-pet",
+            "fake-workspace-user-facing-id",
+            mockCrlService,
+            fakeCliConfiguration,
+            fakeVersionConfig);
+
+    final FlightMap inputFlightMap = new FlightMap();
+    inputFlightMap.put(
+        WorkspaceFlightMapKeys.ControlledResourceKeys.CREATE_NOTEBOOK_PARAMETERS,
+        ControlledResourceFixtures.defaultNotebookCreationParameters());
+    inputFlightMap.makeImmutable();
+    doReturn(inputFlightMap).when(mockFlightContext).getInputParameters();
+
+    final FlightMap workingFlightMap = new FlightMap();
+    GcpCloudContext fakeCloudContext = new GcpCloudContext(FAKE_PROJECT_ID);
+
+    workingFlightMap.put(
+        WorkspaceFlightMapKeys.ControlledResourceKeys.GCP_CLOUD_CONTEXT, fakeCloudContext);
+    workingFlightMap.put(CREATE_NOTEBOOK_LOCATION, "fake-location");
+    doReturn(workingFlightMap).when(mockFlightContext).getWorkingMap();
+
+    // Mock the permission error thrown when polling the error (e.g., quota exceeded limit)
+    when(mockOperationAdapter.getError()).thenReturn(mockOperationStatusAdapter);
+    when(mockOperationStatusAdapter.getCode()).thenReturn(Code.PERMISSION_DENIED_VALUE);
```

Comment:
all the set up before this part is not really necessary. can we just test this part in GcpUtilsTest? sorry i don't think GcputilsTest doesn't exists yet -.-

Reply:
Sure! Will do.

Goodbye cow 🐄 . \o/

Reply:
pollAndRetry still takes in OperationCow so i don't  think you can completely say goodbye to it

Reply:
:(

Welcome back cow!